ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/helpers/LikeEvaluatorTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.codegen.helpers;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+import java.util.Objects;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class LikeEvaluatorTest {
+
+  private final String name;
+  private final String val;
+  private final String pattern;
+
+  @Parameters(name = "{0}")
+  public static Object[][] data() {
+    // name | val | pattern
+    return new Object[][]{
+        {"nothing special",               "foo", "foo"},
+        {"nothing special [X]",           "bar", "foo"},
+        {"empty percents",                "foo", "%foo%"},
+        {"empty percents [X]",            "bar", "%foo%"},
+        {"percents",                      "barfoobaz", "%foo%"},
+        {"percents [X]",                  "barbarbar", "%foo%"},
+        {"percents one side",             "barfoo", "%foo"},
+        {"percents one side [X]",         "barbarbar", "%foo"},
```

Comment:
```suggestion
        {"percents one side [X]",         "barfoobar", "%foo"},
```

Reply:
also, worth adding case for:
*  % other side and 
* % in middle and 
* multiple %s
* escaped %s???? Or do we not support them yet.. (hard to add?)

Reply:
the middle two are covered by `c[a-c]t%[a-e][^abc]%m__w` - i'll add tests for the first and last (escaping is done by `[%]` in MSFT-sql at least and should be supported here as well)